libs/ui-components/src/utils/status/fleet.ts

```diff
@@ -44,4 +47,15 @@ const getFleetSyncStatus = (
   };
 };
 
-export { getFleetSyncStatus, fleetStatusLabels };
+const isFleetRolloutFailedCondition = (rolloutCondition: Condition) =>
+  rolloutCondition.status === ConditionStatus.ConditionStatusFalse &&
+  rolloutCondition.reason === FLEET_ROLLOUT_FAILED_REASON;
+
+export const getFleetRolloutStatusWarning = (fleet: Fleet, t: TFunction) => {
+  const rolloutCondition = fleet.status?.conditions?.find((c) => c.type === ConditionType.FleetRolloutInProgress);
+
+  if (rolloutCondition && isFleetRolloutFailedCondition(rolloutCondition)) {
+    return getConditionMessage(rolloutCondition) || t('Last rollout did not complete successfully');
+  }
+  return undefined;
+};
```

Comment:
Could we refactor this so that the condition selectors are in one place ? It makes it a bit easier to understand. WDYT ?

```suggestion
const isFleetRolloutFailedCondition = (condition: Condition) =>
  condition.type === ConditionType.FleetRolloutInProgress &&
  condition.status === ConditionStatus.ConditionStatusFalse &&
  condition.reason === FLEET_ROLLOUT_FAILED_REASON;

export const getFleetRolloutStatusWarning = (fleet: Fleet, t: TFunction) => {
  const failedRolloutCondition = fleet.status?.conditions?.find(isFleetRolloutFailedCondition);

  if (failedRolloutCondition) {
    return getConditionMessage(failedRolloutCondition) || t('Last rollout did not complete successfully');
  }
  return undefined;
};
```